### DISCUSSION TITLE
artefacts

### DISCUSSION THREAD

**Discussion by liujiynagchian** [QSMxT/QSMxT#135] [General]
Hello everyone, and thank you very much for providing the software.
After running qsmxt to process my data(Philips)，there are many artefacts in the qsm map .
Is there any way to solve this problem.
![image](https://github.com/user-attachments/assets/11c015e6-3ed2-414f-9906-94cbabce112a)

**Comment by liujiynagchian** [comment]
The exmaple data is divided into 2 files and uploaded
[data_part1.zip](https://github.com/user-attachments/files/17095635/data_part1.zip)
[data_part2.zip](https://github.com/user-attachments/files/17095637/data_part2.zip)

**Reply by astewartau** [reply]
Thanks for sharing the data! It looks possibly like issues caused by the masking parameter defaults being not quite right for your data. I can check soon and get back to you. I would first try running with a different `--premade` option to see if the result is any different. The `--premade bet` option will use a BET-derived mask which cannot contain the kinds of holes that I would think lead to the artefact seen.

**Reply by liujiynagchian** [reply]
I have already used the --premade bet option, but there are still artifacts

**Reply by astewartau** [reply]
Thanks for the information! I've had a closer look and found that this artefact comes up with all QSM algorithms in QSMxT. It seems to start at the unwrapping step:

![image](https://github.com/user-attachments/assets/ab60f195-239d-4e3f-b336-a4642b7e9c63)

@korbinian90 - I got these artefacts when running ROMEO generating a field map using [this script](https://github.com/QSMxT/QSMxT/blob/main/qsmxt/scripts/romeo_unwrapping.jl) with the data @liujiynagchian uploaded and command:

```bash
romeo_unwrapping.jl \
    --no-rescale \
    --compute-B0 B0.nii \
    --phase ....../multi-echo-phase.nii \
    --mag ...../multi-echo-mag.nii \
    -t '[5.0 11.0 17.0 23.0 29.0 35.0]' \
    --phase-offset-correction bipolar
```

Have you seen anything like this before? 🤔

**Reply by korbinian90** [reply]
That is weird, I couldn't reproduce it. Here is my B0.nii result
![image](https://github.com/user-attachments/assets/feb53d91-09a7-420b-8e34-c0c5a427bb9a)

What version of romeo is running for you? It is in the settings_romeo.txt output

This is the command that I was running, it should be identical:
`romeo_unwrapping.jl --no-rescale --compute-B0 B0.nii --phase phase.nii.gz --mag mag.nii.gz -t '[5.0 11.0 17.0 23.0 29.0 35.0]' --phase-offset-correction bipolar`

I couldn't find any significant change that I did for ROMEO in over a year. Still you could try to run this slightly modified script in a new folder, it should download the newest package versions: https://github.com/korbinian90/ROMEO.jl/blob/master/romeo.jl
(The script from before used packages that were installed in the global julia environment)

I used it in a new folder and it installed
  [557dad86] + MriResearchTools v3.1.6
  [1ea8258b] + ROMEO v1.2.1

**Reply by astewartau** [reply]
Hi @korbinian90, it's been a while! I hope things are well with you. I have some time to look at this, and the issue has possibly come up again with another user. I ran this with Julia v1.9.3 and still had the same issue. This is my romeo_settings.txt:

```
(qsmxt) ~/downloads/romeo-bug: cat settings_romeo.txt 
mmap-mag: true
compute-B0: B0.nii
output: 
mask-unwrapped: false
weights: romeo3
max-seeds: 1
B0-phase-weighting: phase_snr
individual-unwrapping: false
no-phase-rescale: true
fix-ge-phase: false
write-quality-all: false
correct-global: false
write-quality: false
temporal-uncertain-unwrapping: 0.0
magnitude: multi-echo-mag.nii
phase-offset-correction: bipolar
no-mmap: false
wrap-addition: 0.0
write-phase-offsets: false
verbose: false
neco: 6
correct-regions: false
merge-regions: false
template: 1
threshold: Inf
mmap-phase: true
multi-channel: false
phase: multi-echo-phase.nii
filename: unwrapped.nii
Arguments: --no-rescale --compute-B0 B0.nii --phase multi-echo-phase.nii --mag multi-echo-mag.nii -t [5.0 11.0 17.0 23.0 29.0 35.0] --phase-offset-correction bipolar
RomeoApp version: App 4.5
```

And this is the script I used:

```julia
#!/usr/bin/env julia
import Pkg
Pkg.activate(@__DIR__)
try
    using ROMEO, MriResearchTools, ArgParse
catch
    Pkg.add(["ROMEO", "MriResearchTools", "ArgParse"])
    using ROMEO, MriResearchTools, ArgParse
end

@time msg = unwrapping_main(ARGS)
println(msg)
```

I installed Julia like this:

```bash
wget https://julialang-s3.julialang.org/bin/linux/x64/1.9/julia-1.9.3-linux-x86_64.tar.gz \
       && tar zxvf julia-1.9.3-linux-x86_64.tar.gz \
       && rm -rf julia-1.9.3-linux-x86_64.tar.gz
```

And produced this output:

![image](https://github.com/user-attachments/assets/6285e74a-bb3d-45cf-a012-4b864fcfab20)

**Comment by korbinian90** [comment]
Hi Ashley,

This is a quite strange problem. I'll try to reproduce it again next week. Thanks for the exact settings.

It can be that temporal unwrapping fails, if the first echo phase is very noisy. You could try it with the additional options:
--individual_unwrapping --correct_global

(This might be slower)

**Reply by astewartau** [reply]
Thanks for the suggestions! I tried with `--individual-unwrapping` and `--correct-global` and that seems to overcome this particular issue (see image below - left is before and right is after). Do you think this is making another important trade-off if I were to set this as a new default? I can also wait to see what you find next week. 

![image](https://github.com/user-attachments/assets/0968a844-0bd8-4bd8-96fa-e630cff234c2)

**Reply by korbinian90** [reply]
We haven't encountered issues yet with this new way. It wouldn't work for disconnected regions (that would require some work).
It's probably more robust in most cases.

(I didn't yet have the time to reproduce the other issue)

**Reply by astewartau** [reply]
No problem, thanks for that! I have updated QSMxT to use this as the default. :)

**Comment by astewartau** [comment]
@liujiynagchian, I realise it has been 5 months since you posted this, but the latest version of QSMxT (v7.3.2) solves the problem you encountered. :)

**Reply by liujiynagchian** [reply]
I really appreciate you letting me know that QSMxT v7.3.2 has fixed the problem
